app/src/main/java/com/hedvig/app/feature/home/ui/changeaddress/ChangeAddressResultActivity.kt

```diff
@@ -0,0 +1,127 @@
+package com.hedvig.app.feature.home.ui.changeaddress
+
+import android.content.Context
+import android.content.Intent
+import android.content.res.Configuration
+import android.os.Bundle
+import android.os.Parcelable
+import androidx.activity.compose.setContent
+import androidx.compose.foundation.layout.Spacer
+import androidx.compose.foundation.layout.size
+import androidx.compose.material.ButtonDefaults
+import androidx.compose.material.Icon
+import androidx.compose.material.MaterialTheme
+import androidx.compose.material.Surface
+import androidx.compose.material.Text
+import androidx.compose.runtime.Composable
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.res.painterResource
+import androidx.compose.ui.res.stringResource
+import androidx.compose.ui.tooling.preview.Preview
+import com.google.android.material.composethemeadapter.MdcTheme
+import com.hedvig.app.BaseActivity
+import com.hedvig.app.R
+import com.hedvig.app.feature.chat.ui.ChatActivity
+import com.hedvig.app.feature.loggedin.ui.LoggedInActivity
+import kotlinx.parcelize.Parcelize
+import java.time.LocalDate
+import java.time.format.DateTimeFormatter
+import java.time.format.FormatStyle
+
+class ChangeAddressResultActivity : BaseActivity() {
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        setContent {
+            ChangeAddressResultScreen(
+                result = intent.getParcelableExtra(RESULT)!!,
+                navigateToHome = { startActivity(LoggedInActivity.newInstance(this, withoutHistory = true)) },
+                navigateToChat = { startActivity(ChatActivity.newInstance(this)) }
+            )
+        }
+    }
+
+    sealed class Result : Parcelable {
+        @Parcelize
+        data class Success(
+            val date: LocalDate,
+        ) : Result()
+
+        @Parcelize
+        object UnderwritingLimitsHit : Result()
+    }
+
+    companion object {
+        private const val RESULT = "RESULT"
+
```

Comment:
To make this view even more general, we could pass a data class containing a drawable res (illustration), title, paragraph and button text + icon. We could then reuse it for some of the intro screens as well. We could also use finish with result and let the caller handle what should happen if the button is pressed. What do you think?

Reply:
My thinking was that the ITPB was generic enough for use in Activities, and hence we did not need to make the Activity reusable 🙂

Reply:
Yeah that makes sense :)